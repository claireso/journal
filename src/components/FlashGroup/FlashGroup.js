import React from 'react'
import PropTypes from 'prop-types'

import Flash from '../Flash'

import * as S from './FlashGroup.styles'

const FlashGroup = ({ messages, ...props }) => {
  if (!messages.length) return null

  return (
    <S.FlashGroupWrapper>
      {messages.map((message, index) => (
        <Flash status={message.status} onClose={props.onClose} key={index} index={index} withBorder={index > 0}>
          {message.message}
        </Flash>
      ))}
    </S.FlashGroupWrapper>
  )
}

FlashGroup.propTypes = {
  messages: PropTypes.arrayOf(
    PropTypes.shape({
      status: PropTypes.string,
      message: PropTypes.string.isRequired
    })
  ).isRequired,
  onClose: PropTypes.func.isRequired
}

export default FlashGroup
