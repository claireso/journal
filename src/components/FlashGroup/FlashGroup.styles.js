import styled from 'styled-components'

export const FlashGroupWrapper = styled.div`
  > div {
    margin: calc(var(--gutter) * -1) calc(var(--gutter) * -1) var(--gutter);
  }
`
