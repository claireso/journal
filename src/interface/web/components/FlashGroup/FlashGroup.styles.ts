import { styled } from '@web/oldtheme'

export const FlashGroupWrapper = styled('div', {
  '> div': {
    margin: '-$5 -$5 $5'
  }
})
