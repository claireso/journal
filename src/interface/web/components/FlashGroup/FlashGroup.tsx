import React from 'react'

import Flash from '../Flash'

export interface Message {
  status: 'default' | 'success' | 'error' | 'info'
  message: string
}

interface FlashGroupProps {
  messages: Message[]
  onClose: (index: number) => void
}

const FlashGroup = ({ messages, ...props }: FlashGroupProps) => {
  if (!messages.length) return null

  return (
    <div>
      {messages.map((message, index) => (
        <Flash status={message.status} onClose={props.onClose} key={index} index={index} withBorder={index > 0}>
          {message.message}
        </Flash>
      ))}
    </div>
  )
}

export default FlashGroup
