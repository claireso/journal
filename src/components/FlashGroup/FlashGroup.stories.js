import React from 'react'
import { action } from '@storybook/addon-actions'

import FlashGroup from './index'

export default {
  title: 'FlashGroup',
  decorators: [(storyFn) => <div style={{ padding: '20px' }}>{storyFn()}</div>]
}

export const Basic = () => {
  const messages = [
    { status: 'default', message: 'Default message' },
    { status: 'success', message: 'Success message' },
    { status: 'error', message: 'Error message' }
  ]

  return <FlashGroup messages={messages} onClose={action('close message')} />
}
