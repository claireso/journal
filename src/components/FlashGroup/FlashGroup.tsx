import React from 'react'

import Flash from '../Flash'

import * as S from './FlashGroup.styles'

interface Message {
  status: 'default' | 'success' | 'error' | 'info'
  message: string
}

interface FlashGroupProps {
  messages: Message[]
  onClose: (index: number) => void
}

const FlashGroup = ({ messages, ...props }: FlashGroupProps) => {
  if (!messages.length) return null

  return (
    <S.FlashGroupWrapper>
      {messages.map((message, index) => (
        <Flash status={message.status} onClose={props.onClose} key={index} index={index} withBorder={index > 0}>
          {message.message}
        </Flash>
      ))}
    </S.FlashGroupWrapper>
  )
}

export default FlashGroup
