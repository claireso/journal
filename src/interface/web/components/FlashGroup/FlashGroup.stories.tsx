import { action } from 'storybook/actions'

import type { Meta, StoryObj } from '@storybook/react-webpack5'

import FlashGroup from './FlashGroup'

const meta: Meta<typeof FlashGroup> = {
  title: 'Components/FlashGroup',
  component: FlashGroup,
  args: {
    onClose: action('close message'),
    messages: [
      { status: 'default', message: 'Default message' },
      { status: 'default', message: 'Default message' },
      { status: 'success', message: 'Success message' },
      { status: 'success', message: 'Success message' },
      { status: 'error', message: 'Error message' },
      { status: 'error', message: 'Error message' },
      { status: 'info', message: 'Info message' },
      { status: 'info', message: 'Info message' }
    ]
  }
}

export default meta
type Story = StoryObj<typeof FlashGroup>

export const Basic: Story = {}
