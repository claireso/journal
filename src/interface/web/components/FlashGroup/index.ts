import FlashGroup from './FlashGroup'

export default FlashGroup
