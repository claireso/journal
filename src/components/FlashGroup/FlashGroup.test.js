import { render, screen, fireEvent } from '@testing-library/react'

import FlashGroup from './index'

describe('<FlashGroup />', () => {
  const messages = [
    {
      status: 'success',
      message: 'Success message'
    },
    {
      status: 'error',
      message: 'Error message'
    }
  ]

  it('should render two messages', () => {
    const { asFragment } = render(<FlashGroup messages={messages} onClose={() => {}} />)

    expect(asFragment()).toMatchSnapshot()
  })

  it('should close messages', () => {
    const onClose = jest.fn()

    const { asFragment } = render(<FlashGroup messages={messages} onClose={onClose} />)

    expect(asFragment()).toMatchSnapshot()

    const buttons = screen.getAllByRole('button')

    fireEvent.click(buttons[0])

    expect(onClose).toHaveBeenNthCalledWith(1, 0)

    fireEvent.click(buttons[1])

    expect(onClose).toHaveBeenNthCalledWith(2, 1)
  })
})
