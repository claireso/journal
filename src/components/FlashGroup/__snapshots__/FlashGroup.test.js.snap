// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FlashGroup /> should close messages 1`] = `
<DocumentFragment>
  <div
    class="c-jVMhMN"
  >
    <div
      class="c-fyLyTf c-fyLyTf-eIkjtV-status-success"
    >
      Success message
      <button
        class="c-ggYOOi c-eUyIYb c-hrWJbv"
      >
        <svg
          height="16"
          viewBox="0 0 22 28"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M20.281 20.656c0 0.391-0.156 0.781-0.438 1.062l-2.125 2.125c-0.281 0.281-0.672 0.438-1.062 0.438s-0.781-0.156-1.062-0.438l-4.594-4.594-4.594 4.594c-0.281 0.281-0.672 0.438-1.062 0.438s-0.781-0.156-1.062-0.438l-2.125-2.125c-0.281-0.281-0.438-0.672-0.438-1.062s0.156-0.781 0.438-1.062l4.594-4.594-4.594-4.594c-0.281-0.281-0.438-0.672-0.438-1.062s0.156-0.781 0.438-1.062l2.125-2.125c0.281-0.281 0.672-0.438 1.062-0.438s0.781 0.156 1.062 0.438l4.594 4.594 4.594-4.594c0.281-0.281 0.672-0.438 1.062-0.438s0.781 0.156 1.062 0.438l2.125 2.125c0.281 0.281 0.438 0.672 0.438 1.062s-0.156 0.781-0.438 1.062l-4.594 4.594 4.594 4.594c0.281 0.281 0.438 0.672 0.438 1.062z"
          />
        </svg>
      </button>
    </div>
    <div
      class="c-fyLyTf c-fyLyTf-epOfjx-status-error c-fyLyTf-bknGQq-separator-true"
    >
      Error message
      <button
        class="c-ggYOOi c-eUyIYb c-hrWJbv"
      >
        <svg
          height="16"
          viewBox="0 0 22 28"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M20.281 20.656c0 0.391-0.156 0.781-0.438 1.062l-2.125 2.125c-0.281 0.281-0.672 0.438-1.062 0.438s-0.781-0.156-1.062-0.438l-4.594-4.594-4.594 4.594c-0.281 0.281-0.672 0.438-1.062 0.438s-0.781-0.156-1.062-0.438l-2.125-2.125c-0.281-0.281-0.438-0.672-0.438-1.062s0.156-0.781 0.438-1.062l4.594-4.594-4.594-4.594c-0.281-0.281-0.438-0.672-0.438-1.062s0.156-0.781 0.438-1.062l2.125-2.125c0.281-0.281 0.672-0.438 1.062-0.438s0.781 0.156 1.062 0.438l4.594 4.594 4.594-4.594c0.281-0.281 0.672-0.438 1.062-0.438s0.781 0.156 1.062 0.438l2.125 2.125c0.281 0.281 0.438 0.672 0.438 1.062s-0.156 0.781-0.438 1.062l-4.594 4.594 4.594 4.594c0.281 0.281 0.438 0.672 0.438 1.062z"
          />
        </svg>
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<FlashGroup /> should render two messages 1`] = `
<DocumentFragment>
  <div
    class="c-jVMhMN"
  >
    <div
      class="c-fyLyTf c-fyLyTf-eIkjtV-status-success"
    >
      Success message
      <button
        class="c-ggYOOi c-eUyIYb c-hrWJbv"
      >
        <svg
          height="16"
          viewBox="0 0 22 28"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M20.281 20.656c0 0.391-0.156 0.781-0.438 1.062l-2.125 2.125c-0.281 0.281-0.672 0.438-1.062 0.438s-0.781-0.156-1.062-0.438l-4.594-4.594-4.594 4.594c-0.281 0.281-0.672 0.438-1.062 0.438s-0.781-0.156-1.062-0.438l-2.125-2.125c-0.281-0.281-0.438-0.672-0.438-1.062s0.156-0.781 0.438-1.062l4.594-4.594-4.594-4.594c-0.281-0.281-0.438-0.672-0.438-1.062s0.156-0.781 0.438-1.062l2.125-2.125c0.281-0.281 0.672-0.438 1.062-0.438s0.781 0.156 1.062 0.438l4.594 4.594 4.594-4.594c0.281-0.281 0.672-0.438 1.062-0.438s0.781 0.156 1.062 0.438l2.125 2.125c0.281 0.281 0.438 0.672 0.438 1.062s-0.156 0.781-0.438 1.062l-4.594 4.594 4.594 4.594c0.281 0.281 0.438 0.672 0.438 1.062z"
          />
        </svg>
      </button>
    </div>
    <div
      class="c-fyLyTf c-fyLyTf-epOfjx-status-error c-fyLyTf-bknGQq-separator-true"
    >
      Error message
      <button
        class="c-ggYOOi c-eUyIYb c-hrWJbv"
      >
        <svg
          height="16"
          viewBox="0 0 22 28"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M20.281 20.656c0 0.391-0.156 0.781-0.438 1.062l-2.125 2.125c-0.281 0.281-0.672 0.438-1.062 0.438s-0.781-0.156-1.062-0.438l-4.594-4.594-4.594 4.594c-0.281 0.281-0.672 0.438-1.062 0.438s-0.781-0.156-1.062-0.438l-2.125-2.125c-0.281-0.281-0.438-0.672-0.438-1.062s0.156-0.781 0.438-1.062l4.594-4.594-4.594-4.594c-0.281-0.281-0.438-0.672-0.438-1.062s0.156-0.781 0.438-1.062l2.125-2.125c0.281-0.281 0.672-0.438 1.062-0.438s0.781 0.156 1.062 0.438l4.594 4.594 4.594-4.594c0.281-0.281 0.672-0.438 1.062-0.438s0.781 0.156 1.062 0.438l2.125 2.125c0.281 0.281 0.438 0.672 0.438 1.062s-0.156 0.781-0.438 1.062l-4.594 4.594 4.594 4.594c0.281 0.281 0.438 0.672 0.438 1.062z"
          />
        </svg>
      </button>
    </div>
  </div>
</DocumentFragment>
`;
